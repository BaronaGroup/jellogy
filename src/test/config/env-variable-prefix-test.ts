import {prepareTestKarhu} from '../test-utils'

describe('env-variable-prefix-test', () => {
  const karhuTest = prepareTestKarhu({
    envVariablePrefix: 'KUHA'
  })

  beforeEach(() => {
    process.env.KUHA_COLOR = '1'
  })

  describe('allows using a different prefix', () => {
    it('applies on COLOR variable', karhuTest((karhu, output) => {
      process.env.KUHA_COLOR = '1'
      karhu.context('ctx').error('should have color')
      process.env.KUHA_COLOR = '0'
      karhu.context('ctx').error('should not have color')
      expect(output.tracked).toMatchSnapshot()
    }))

    it('applies to default log level override', karhuTest((karhu, output) => {
      process.env.KUHA_LOG_LEVEL = 'WARN'
      karhu.context('ctx').warn('should show up')
      process.env.KUHA_LOG_LEVEL = 'ERROR'
      karhu.context('ctx').warn('should not show up')
      process.env.KUHA_LOG_LEVEL = ''
      expect(output.tracked).toMatchSnapshot()
    }))

    it('applies to context specific overrides', karhuTest((karhu, output) => {
      process.env.KUHA_LOG_LEVEL_highprio = 'DEBUG'
      karhu.context('basic').debug('should not show up')
      karhu.context('highprio').debug('should show up')
      expect(output.tracked).toMatchSnapshot()
    }))
  })
})
