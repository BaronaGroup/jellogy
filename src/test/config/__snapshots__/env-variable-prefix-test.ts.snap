// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`env-variable-prefix-test allows using a different prefix applies on COLOR variable 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [
    Array [
      "0 ERROR ctx[97m[101m",
      "should have color",
      "[49m[39m",
    ],
    Array [
      "1 ERROR ctx",
      "should not have color",
    ],
  ],
}
`;

exports[`env-variable-prefix-test allows using a different prefix applies to context specific overrides 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [
    Array [
      "0 DEBUG highprio[95m",
      "should show up",
      "[39m",
    ],
  ],
  "INFO" => Array [],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;

exports[`env-variable-prefix-test allows using a different prefix applies to default log level override 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [],
  "WARN" => Array [
    Array [
      "0 WARN ctx[93m",
      "should show up",
      "[39m",
    ],
  ],
  "ERROR" => Array [],
}
`;
